### PR TITLE
Add xvnc, ssh to supportserver roles and 4 remote desktop cases to x11regressions

### PR DESCRIPTION
**Pull request by chuchingkai**
## Update supportserver boot, login and add xvnc, ssh to supportserver roles

- We are going to implement remote desktop testing in poo#9504 which needs:
    * SLES12SP2+gnome(gdm/vino...is necessary) supportserver qcow2 image
    * one time session xvnc server
    * persistent session xvnc server
    * ssh server for X11 forwarding
    * vino server
    * XDMCP with gdm and SLE-Classic configured
    * XDMCP with xdm and icewm configured
    * xrdp server
  supportserver is a good choice for Remote Desktop testing, but we have to
  update the login part and add the needed server roles in setup.pm.
- We used to use boot.pm and login.pm to boot into the supportserver image and
  it only supports console login. boot.pm is dropped in this commit and login.pm
  is updated by using wait_boot which supports both console and graphical login.
- Adding xvnc server role which supports both onetime VNC session and persistent
  VNC session, the VNC type is configured by variable in testsuits:
  REMOTE_DESKTOP_TYPE = persistent_vnc
  REMOTE_DESKTOP_TYPE = one_time_vnc

## Add 4 remote desktop cases for x11regressions

- onetime_vncsession_xvnc_java.pm
  Remote Login: One-time VNC Session with Jave applet and xvnc
- onetime_vncsession_xvnc_tigervnc.pm
  Remote Login: One-time VNC Session with tigervnc and xvnc
- persistent_vncsession_xvnc.pm
  Remote Login: Persistent VNC Session with tigervnc and xvnc
- x11_forwarding_openssh.pm
  Remote Login: X11 forwarding over OpenSSH

see also: https://progress.opensuse.org/issues/9504

Validation runs:
remote-desktop-client1: http://147.2.212.167/tests/298
remote-desktop-supportserver1: http://147.2.212.167/tests/296
remote-desktop-client2: http://147.2.212.167/tests/299
remote-desktop-supportserver2: http://147.2.212.167/tests/297
supportserver regression test: http://147.2.212.167/tests/295

### CODE REVIEW

File: tests/support_server/login.pm [hunk 1]
```diff
@@ -13,34 +13,24 @@
 # You should have received a copy of the GNU General Public License along
 # with this program; if not, see <http://www.gnu.org/licenses/>.
 
-# G-Summary: supportserver and supportserver generator implementation
-# G-Maintainer: Pavel Sladek <psladek@suse.com>
+# Summary: supportserver and supportserver generator implementation
```
Comment (okurz) [inline tests/support_server/login.pm]: please add a better summary for this specific test module

File: tests/x11regressions/remote_desktop/onetime_vncsession_xvnc_java.pm [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    assert_and_click "xvnc-firefox-trust-ca";
+    assert_screen "firefox-java-security";
+    assert_and_click "firefox-java-securityrun";
+    assert_screen "xvnc-firefox-verification-warning";
+    send_key "ret";
+    if (check_screen 'xvnc-firefox-certification-warning2') {
+        send_key "ret";
+    }
+    assert_and_click "xvnc-firefox-dm", 90;
+    sleep 2;
```
Comment (okurz) [inline tests/x11regressions/remote_desktop/onetime_vncsession_xvnc_java.pm]: why?

File: tests/x11regressions/remote_desktop/persistent_vncsession_xvnc.pm [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+use utils;
+
+sub start_vncviewer {
+    x11_start_program "vncviewer 10.0.2.1:1 -Fullscreen";
+    if (check_screen 'vnc_certificate_warning', 10) {
+        send_key "ret";
+    }
+    if (check_screen 'vnc_certificate_warning-2', 10) {
+        send_key "ret";
+    }
```
Comment (okurz) [inline tests/x11regressions/remote_desktop/persistent_vncsession_xvnc.pm]: better not use `check_screen` with a low timeout value but `assert_screen` on multiple tags and `match_has_tag`

**Comment by chuchingkai** [comment]
> the supportserver change might be a bit intrusive
@okurz I also have this concern but I'm not familiar with support server. So I have done a regression test to make sure the commit doesn't break other support server test:
supportserver regression test: http://147.2.212.167/tests/295

If this change is not suitable, I can separate my new code from the from the original ones, make everything under x11regressions/remote_desktop

**Comment by okurz** [comment]
thank you for testing supportserver more.

LGTM then

File: tests/x11regressions/remote_desktop/onetime_vncsession_xvnc_java.pm [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    assert_and_click 'xvnc-firefox-IcedTea-allow';
+    assert_screen 'xvnc-firefox-certification-warning';
+    assert_and_click 'xvnc-firefox-trust-ca';
+    assert_screen 'firefox-java-security';
+    assert_and_click 'firefox-java-securityrun';
+    assert_screen 'xvnc-firefox-verification-warning';
+    send_key 'ret';
+    if (check_screen 'xvnc-firefox-certification-warning2') {
+        send_key 'ret';
+    }
```
Comment (okurz) [inline tests/x11regressions/remote_desktop/onetime_vncsession_xvnc_java.pm]: IMHO you can replace by

```perl
assert_screen [qw(xvnc-firefox-certification-warning2 xvnc-firefox-dm)];
send_key 'ret' if (match_has_tag 'xvnc-firefox-certification-warning2');
```
and then continue with the next line to save waiting time.

Comment (chuchingkai) [inline tests/x11regressions/remote_desktop/onetime_vncsession_xvnc_java.pm]: fixed

File: tests/x11regressions/remote_desktop/onetime_vncsession_xvnc_java.pm [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    assert_and_click 'xvnc-firefox-activate-IcedTea';
+    assert_and_click 'xvnc-firefox-IcedTea-allow';
+    assert_screen 'xvnc-firefox-certification-warning';
+    assert_and_click 'xvnc-firefox-trust-ca';
+    assert_screen 'firefox-java-security';
+    assert_and_click 'firefox-java-securityrun';
+    assert_screen 'xvnc-firefox-verification-warning';
+    send_key 'ret';
+    assert_and_click [qw(xvnc-firefox-certification-warning2 xvnc-firefox-dm)];
+    if (match_has_tag 'xvnc-firefox-certification-warning2') {
```
Comment (okurz) [inline tests/x11regressions/remote_desktop/onetime_vncsession_xvnc_java.pm]: just to be sure. Did you check this is working after the adaption?

Comment (chuchingkai) [inline tests/x11regressions/remote_desktop/onetime_vncsession_xvnc_java.pm]: Yes, http://147.2.212.167/tests/331

**Comment by okurz** [comment]
thank you

**Pull request merged**